src/core/grant-authorization.ts

```diff
@@ -15,10 +15,16 @@ export class GrantAuthorization {
   public static async authorizeGenericMessage(
     tenant: string,
     incomingMessage: Message<GenericMessage>,
-    author: string,
     messageStore: MessageStore,
-  ): Promise<void> {
+  ): Promise<PermissionsGrantMessage> {
     const permissionsGrantId: string = incomingMessage.authorizationPayload!.permissionsGrantId!;
+    if (incomingMessage.author === undefined) {
```

Comment:
I am supportive for defensive check here as a fall back check. But what do you think of the pattern that I introduced where in general auth methods aren't called for unauthenticated calls? I find the code much easier to reason about. If you agree I can make a similar change that I did with `ProtocolsQuery` and `RecordsQuery` to `RecordsRead` in a future PR (I just forgot its existence at the time).

Reply:
Updated. I structured it slightly different than in `RecordsQuery` and `ProtocolsQuery`, but we can make it exactly the same in a future PR if you think that's better.